File: demo-material3/lib/features/peer_evaluation/domain/models/member_result.dart
// Pure Dart — NO Flutter imports
import 'peer.dart';

class MemberResult {
  final Peer peer;
  /// criteriaLabel → average score
  final Map<String, double> scores;

  const MemberResult({
    required this.peer,
    required this.scores,
  });

  double get average {
    if (scores.isEmpty) return 0.0;
    final total = scores.values.fold(0.0, (sum, s) => sum + s);
    return total / scores.length;
  }
}

File: demo-liquid-glass/DomainKit/Sources/DomainKit/domain/models/MemberResult.swift
// domain/models/MemberResult.swift
// Pure Swift struct — NO SwiftUI, NO Foundation imports

/// Aggregated evaluation result for a single peer across all evaluators.
public struct MemberResult: Identifiable, Equatable {
    public let id: String
    public let peer: Peer
    /// key: criteria name, value: average score across all evaluators
    public let scores: [String: Double]

    public init(peer: Peer, scores: [String: Double]) {
        self.id = peer.id
        self.peer = peer
        self.scores = scores
    }

    /// Overall average score across all criteria.
    public var average: Double {
        guard !scores.isEmpty else { return 0.0 }
        let total = scores.values.reduce(0.0, +)
        return total / Double(scores.count)
    }
}
